go.sum

```diff
@@ -720,16 +813,27 @@ google.golang.org/api v0.14.0/go.mod h1:iLdEw5Ide6rF15KTC1Kkl0iskquN2gFfn9o9XIsb
 google.golang.org/api v0.15.0/go.mod h1:iLdEw5Ide6rF15KTC1Kkl0iskquN2gFfn9o9XIsbkAI=
 google.golang.org/api v0.17.0/go.mod h1:BwFmGc8tA3vsd7r/7kR8DY7iEEGSU04BFxCo5jP/sfE=
 google.golang.org/api v0.18.0/go.mod h1:BwFmGc8tA3vsd7r/7kR8DY7iEEGSU04BFxCo5jP/sfE=
+google.golang.org/api v0.19.0/go.mod h1:BwFmGc8tA3vsd7r/7kR8DY7iEEGSU04BFxCo5jP/sfE=
 google.golang.org/api v0.20.0/go.mod h1:BwFmGc8tA3vsd7r/7kR8DY7iEEGSU04BFxCo5jP/sfE=
+google.golang.org/api v0.22.0/go.mod h1:BwFmGc8tA3vsd7r/7kR8DY7iEEGSU04BFxCo5jP/sfE=
```

Comment:
There might be some things in here which are unrelated to the current change or are the result of `go mod tidy`. I think you can make the change to this file smaller if you do something like:
```sh
git checkout v1 -- go.*
go get cloud.google.com/go/pubsub
git add go.* && git commit
```

Reply:
I did this, but the CI broke again.

```
# gopkg.in/DataDog/dd-trace-go.v1/contrib/google.golang.org/api [gopkg.in/DataDog/dd-trace-go.v1/contrib/google.golang.org/api.test]
contrib/google.golang.org/api/api_test.go:73:30: too many arguments in call to svc.Representatives.RepresentativeInfoByAddress
contrib/google.golang.org/api/api_test.go:73:32: undefined: civicinfo.RepresentativeInfoRequest
FAIL	gopkg.in/DataDog/dd-trace-go.v1/contrib/google.golang.org/api [build failed]
```

I suspect that since go.mod is not being used correctly in this repo right now (you can test this trivially by checking out `v1`, running `go test ./contrib/...` and noticing that `go.mod`, that is basically empty right now, gets correctly populated with all the direct/indirect dependencies; that should not happen if `go.mod` had been correctly committed)

Reply:
> I suspect that since go.mod is not being used correctly in this repo right now [...]

I wouldn't say it's not used correctly. I would say it's not used conventionally as you are used to it :) There is a reason why `go.mod` is empty. I'll try to explain. Perhaps I've omitted something somewhere and you can help me figure it out.

Here is an example use-case illustrating a problem:
1. A user is using the `pubsub@v1.4.0` in his application
2. User gets `dd-trace-go` to trace his `pubsub@v1.4.0` library.
3. Getting `dd-trace-go` which has a `pubsub@v1.6.1` in its `go.mod` will force the user's build to update and use the most recent version, which is `v1.6.1`.

Do you see the problem? In some cases this can be critical, introducing new bugs and problems. See "[version selection](https://github.com/golang/go/wiki/Modules#version-selection)" on the Go wiki.

That's the reason why `go.mod` is empty. We don't want to override anything in the user's system. We want the application as a whole to continue using what the user selects. If we enforce versions, it can have serious adverse effects and step on people's toes.

Now, I might have omitted something not being an expert in Go modules, if so, happy to change this way and implement it differently.

P.S. For this reason development is a bit tricky too. One has to ignore `go.mod` when committing into the repository. So please remove the changes to `go.mod` from the PR (unless we find a better way to solve the above together).

Reply:
The problem that you are having is not because of our `go.mod` or CI, it is because of breaking changes in v0 libraries. To fix it, we need to add a line just before running the `google.golang.org/api` tests:

```sh
go get google.golang.org/api@v0.28.0 # breaking change introduced after this patch
```

That should do it. This is the peril of adding integrations for v0 libraries. Lesson learned. It's ugly, I know, but we haven't found a better way and are open to suggestions.

Reply:
Thanks for the help here. 

Regarding go.mod, I get where this is coming from but I think this solution does not really protect you very much. While it is true that specifying a new version in go.mod will cause users to potentially have the version they use bumped as well, consider:

- dd-trace-go runs CI tests, obviously against a certain version X of each dependency. Not specifying the version in go.mod does not give the user any hint about which version X was actually tested. As a user this is valuable information.
- Specifying a version in your go.mod does not "enforce" anything really. Users can still override the version on their side, if they choose so for any reason, using `go get` or - in extreme cases - with `replace`.
- In this case for example, a user with 1.4.0 would have still had their build broken (because 1.4.0 was missing some symbols used by this module) and go mod would have been unable to fix the problem automatically - since the go.mod in this repo was missing the indication that 1.6.1 was required.
- If two contrib modules required different versions of a common dependency, using go.mod will prevent you from hopelessly breaking users' build that were to use both contrib modules

Dependency version selection is a problem with imperfect solutions and tons of tradeoffs, that much is clear. But adding complexity to the process by not working with the existing tools is unlikely to make for a better UX.

Anyway we're veering way too much off course here. I appreciate the help in fixing the build!

Reply:
You have good points. We have considered all of that, and have discussed this internally to great extents. Not to say that I'm not open to discussing it again. If you want to jump in on the discussion we should open an issue and take it there.

FWIW the main focus is to be as unobtrusive to users as possible. The focus is on the user, even if the development process takes a bit of a beating sometimes.

We haven't yet received any such issues but if we do, we'll re-evaluate to find better ways.